### PR TITLE
Fix classes applied to different UserBadge sizes

### PR DESCRIPTION
**Pull request by nickgros**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/widget/discussion/DiscussionThreadListItemWidget.java [hunk 1]
```diff
@@ -39,12 +40,12 @@ public void configure(DiscussionThreadBundle bundle) {
...
 		view.setTitle(bundle.getTitle());
 		view.setThreadUrl(TopicUtils.buildThreadLink(bundle.getProjectId(), bundle.getId()));
-		authorWidget.setTextHidden(true);
+		authorWidget.setBadgeType(BadgeType.SMALL_CARD);
 		authorWidget.configure(bundle.getCreatedBy());
 		view.clearActiveAuthors();
 		for (String userId : bundle.getActiveAuthors()) {
 			UserBadge user = ginInjector.getUserBadgeWidget();
-			user.setTextHidden(true);
+			user.setBadgeType(BadgeType.AVATAR);
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/discussion/DiscussionThreadListItemWidget.java:48]: This looks kind of odd in production, here it is after changes: 

image

File: src/main/java/org/sagebionetworks/web/client/presenter/PeopleSearchPresenter.java [hunk 2]
```diff
@@ -87,7 +87,7 @@ public void onSuccess(UserGroupHeaderResponsePage result) {
 				for (UserGroupHeader header : result.getChildren()) {
 					if (header.getIsIndividual()) {
 						UserBadge badge = ginInjector.getUserBadgeWidget();
-						badge.setSize(BadgeSize.MEDIUM);
+						badge.setBadgeType(BadgeType.MEDIUM_CARD);
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/presenter/PeopleSearchPresenter.java:90]: image

File: src/main/java/org/sagebionetworks/web/client/widget/discussion/ReplyWidget.java [hunk 3]
```diff
@@ -93,6 +95,9 @@ public void configure(DiscussionReplyBundle bundle, Boolean isCurrentUserModerat
 		this.moderatorIds = moderatorIds;
 		this.deleteReplyCallback = deleteReplyCallback;
 		this.isThreadDeleted = isThreadDeleted;
+		authorWidget.setBadgeType(BadgeType.SMALL_CARD);
+		authorWidget.setShowAvatar(true);
+		authorWidget.setAvatarSize(AvatarSize.MEDIUM);
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/discussion/ReplyWidget.java:100]: image

File: src/main/java/org/sagebionetworks/web/client/widget/discussion/SingleDiscussionThreadWidget.java [hunk 4]
```diff
@@ -186,7 +188,9 @@ public void configure(DiscussionThreadBundle bundle, String replyId, Boolean isC
 		configureView(bundle);
 		boolean isAuthorModerator = moderatorIds.contains(bundle.getCreatedBy());
 		view.setIsAuthorModerator(isAuthorModerator);
-
+		authorWidget.setBadgeType(BadgeType.SMALL_CARD);
+		authorWidget.setShowAvatar(true);
+		authorWidget.setAvatarSize(AvatarSize.MEDIUM);
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/discussion/SingleDiscussionThreadWidget.java:193]: image

File: src/main/java/org/sagebionetworks/web/client/widget/header/HeaderViewImpl.java [hunk 5]
```diff
@@ -147,8 +148,9 @@ public HeaderViewImpl(Binder binder, SageImageBundle sageImageBundle, SearchBox
 		this.sageImageBundle = sageImageBundle;
 		this.userBadge = userBadge;
 		this.globalAppState = globalAppState;
-		userBadge.setTextHidden(true);
-		userBadge.setSize(BadgeSize.SMALL_AVATAR);
+		userBadge.setBadgeType(BadgeType.AVATAR);
+		userBadge.setAvatarSize(AvatarSize.MEDIUM);
+		userBadge.setShowCardOnHover(false);
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/header/HeaderViewImpl.java:153]: image

File: src/main/java/org/sagebionetworks/web/client/widget/team/MemberListWidgetViewImpl.java [hunk 6]
```diff
@@ -55,7 +55,7 @@ public void addMembers(List<TeamMemberBundle> members, boolean isAdmin) {
 			singleGridItem.addStyleName("SRC-grid-item");
 			final UserProfile member = teamMember.getUserProfile();
 			UserBadge userBadge = portalGinInjector.getUserBadgeWidget();
-			userBadge.setSize(BadgeSize.MEDIUM);
+			userBadge.setBadgeType(BadgeType.MEDIUM_CARD);
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/team/MemberListWidgetViewImpl.java:58]: image

**Review by xschildw — APPROVED** [review]

**Pull request merged**